### PR TITLE
Remove deprecated variadic listen

### PR DESCRIPTION
**Pull request by jsumners**
So much nasty code gone.

deleting-code

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Uzlopak — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14, 16, 18, 20]
+        node-version: [16, 18, 20]
         os: [macos-latest, ubuntu-latest, windows-latest]
         exclude:
```
Comment (climba03003) [inline .github/workflows/ci.yml:82]: If the node version need to be updated, `exclude` should be removed. And I will eagerly remove both `14` and `16` at the same time with another PR.

**Review by climba03003 — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jsumners) [reply .github/workflows/ci.yml]: FYI, I was only removing enough for tests to pass. It is known that 14 will not be supported in v5. I _assume_ we will be releasing v5 late enough that 16 will not be supported, but it is not guaranteed yet. So as part of the v5 prep process a PR should be created to update CI specifically.

**Pull request merged**